lib/phoenix_live_view/html_formatter.ex

```diff
@@ -554,6 +556,22 @@ defmodule Phoenix.LiveView.HTMLFormatter do
 
   defp may_set_preserve(buffer, _text), do: buffer
 
+  defp may_set_preserve_on_text([{:text, text, meta}], tag_name)
+       when tag_name in @inline_elements and tag_name != "button" do
```

Comment:
I'm doing exclusively for inline elements and excluding buttons. Maybe there are other tags in the inline elements that we  could potentially exclude. Let me know if that makes sense 🙌🏻